### ISSUE TITLE
question about the batch_size in training lora

### ISSUE THREAD

**Issue by xinli2008** [kohya-ss/sd-scripts#1273]
Thank you for your nice job! I'm currently using your repository to train LoRa, but I noticed that in the actual code, the batch_size option seems to be fixed as 1, as shown in the following figure.

![image](https://github.com/kohya-ss/sd-scripts/assets/71311087/b0c2c6c4-ccd0-421a-9b34-843735576c49)

Because my machine has 48GB of RAM, and I have a large amount of data, I want to increase the batch_size. Can I simply adjust the batch_size of the dataloader, or do I need to make other adjustments as well?

Best regards

**Comment by xinli2008** [comment]
Furthermore, I'm puzzled because in my understanding, batch_size works in conjunction with the dataloader. However, in this code, the batch_size in the dataloader is fixed. So, can I assume that it doesn't matter what value I set for the train_batch_size parameter?

**Comment by DKnight54** [comment]
Can I clarify, when you say you want to increase the batch size, I do assume you mean the batch size of the images in once batch of training step?

Also, not really knowing your training enviroment, I'm assuming that you are training on a GPU and not a CPU, as the batch size during training mainly affects GPU VRAM usage, and not RAM.

48 GB of VRAM is not impossible if you are using a professional GPU(such as a [RTX A6000](https://www.nvidia.com/en-us/design-visualization/rtx-a6000/)) instead of a consumer GPU, which maxes out at 24 GB VRAM, so I'm not going to make any assumptions.

While I don't fully understand the code behind the dataloaders, based on my personal experience on running trainings, I can assure you that the train_batch_size does affect the number of images being loaded per training step, as increasing the value does changes how much VRAM is being use. Push the batch number too high, and you'd run out of VRAM and get the OOM error, or experience a dramatic slowdown in training speed as if the loaded models, weights, gradients and latents(images) exceed the VRAM, they can overflow to the RAM which prevents OOM errors, but slows down the training due to the additional latency from loading data in and out of RAM.

I suggest you test for yourself by trying different values for train_batch_size and see how much VRAM is being used. Don't expect dramatic changes, as the loaded models and LoRA weights will also take up a significant portion of the VRAM.

Do note, if you are running training on a CPU, then it will be using RAM instead of VRAM.

**Comment by BootsofLagrangian** [comment]
sd-scripts uses `double-dataloader` for Aspect Ratio Bucketing(ARB).  `The major dataloader` try to pick up a bucket from ARB, which has `batch size = 1 `because size of bucket is always one. 

`The minor dataloader` try to collect image of batch size ( they use args.train_batch_size ).

As you can see, `train_dataloader` is dataloader for ARB. Still you can set a larger batch size because it works on `inside of Bucket`.
